### PR TITLE
[WIP] Port /vg/ improvements to SDQL

### PR DESCRIPTION
**Pull request by neersighted**
This allows arrays in queries, eg:
UPDATE /mob/living SET faction = ["neutral", "silicon"]

As well as switching to absolute paths

### CODE REVIEW

**Comment by hornygranny** [comment]
sweet

**Comment by JJRcop** [comment]
um did you kill #4940?

**Comment by MrStonedOne** [comment]
ya, killing mutiquery support is lame.

also, this shouldn't be merged until you fix the exploit allowing you to edit restricted vars

Their sdql2 allows doing update thing set var.var.var = blah where blah, so you can do update mob set client.holder.flags = 65535 where ckey = "mrstonedone"

That bug is why i never ported it.

**Comment by neersighted** [comment]
Ah, was unaware of these two things. I'll see what I can do.

File: code/modules/admin/verbs/SDQL2/SDQL_2.dm [hunk 1]
```diff
@@ -1,6 +1,4 @@
-// Code taken from /bay/station.
-// Modified to allow consequtive querys in one invocation, terminated with ";"
```
Comment (JJRcop) [inline code/modules/admin/verbs/SDQL2/SDQL_2.dm]: forgive me for the salt, but, "unaware"?

**Comment by neersighted** [comment]
@JJRcop Nothing personal, it was late at night and mostly copypasta. Like I said I'm going to hand merge your improvements.

**Comment by JJRcop** [comment]
@neersighted Ah, okay. I can understand that completely thanks to #2277, you're good.

**Comment by neersighted** [comment]
WIP until I can merge the fixes. @JJRcop PRs are welcome.

**Comment by JJRcop** [comment]
Alr*iiiiight* I *GUEEEESSSSS* I can break my sort-of-monthly-but-not-all-the-time coding tradition and I *GUEEESSS* I can redo my stuff. But it will have to be tomorrow, I have an 8 hr job.

(Also I'm joking I want to code more often but I can't seem to break my on/off habit.)

**Comment by JJRcop** [comment]
My deepest apologies, I will work on it later on today.

**Comment by JJRcop** [comment]
um

**Comment by neersighted** [comment]
**memes**

**Comment by tkdrg** [comment]
@neersighted are you still working on this

**Comment by JJRcop** [comment]
i think he was waiting on me

**Comment by neersighted** [comment]
Yeah, waiting on JJR...

**Comment by JJRcop** [comment]
ok um, sorry, i will now

**Comment by JJRcop** [comment]
i wish i looked at this earlier, i really do, my job took a lot of my time. but i'm sorry, this wasn't the two-code-block revert i thought it was, it's a complete reversal, is it really my place to completely redo everything *I already did* two years ago? who else has done that?

**Comment by tkdrg** [comment]
@neersighted do you still plan on working on this

**Comment by neersighted** [comment]
go ahead and close it; I'll re-open later

**Comment by PJB3005** [comment]
@MrStonedOne 

> Their sdql2 allows doing update thing set var.var.var = blah where blah, so you can do update mob set client.holder.flags = 65535 where ckey = "mrstonedone"

No you can't. I added a specific exception for that.

You don't know what you're missing out on by not having SDQL, it's by far the best admin tool for both debugging and admin bus.

**Comment by MrStonedOne** [comment]
we do have sdql2, just not a few things you guy have added.

**Pull request closed without merging**